### PR TITLE
kodi.packages.arteplussept: init at 1.1.1

### PR DESCRIPTION
**Pull request by dschrempf**
Add the Arte+7 Kodi plugin.

This PR also includes a tiny fix for the Kodi ORF TVThek plugin (add +matrix to version string for compatibility; add automatic updates). I did not open a separate PR for this, because I thought it is easier like so, but please let me know, if you require a separate PR.

Ping @aanderse for review.

I tested the plugins and they work on my Linux machine.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by aanderse — CHANGES_REQUESTED** [review]
Thanks for the PR!

> add automatic updates

Anything which has releases on github should automatically be taken care of by the update bot, so you can drop the change to the `orftvthek` addon.

> if you require a separate PR

One PR is fine, but please split your changes up into different commits.

> I tested the plugins and they work on my Linux machine.

Great! Code looks fine :+1: ... though my personal opinion would be to omit the whitespace changes please.

**Comment by dschrempf** [comment]
Thank you.

I split the commits, removed the update script wherever possible, and removed the whitespace changes, which had been introduced by the automatic formatter.

File: pkgs/applications/video/kodi/addons/xbmcswift2/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  };
+
+  passthru = {
+    pythonPath = "lib";
+  };
+
+  meta = with lib; {
+    homepage = "https://github.com/XBMC-Addons/script.module.xbmcswift2";
+    description = "Framework to ease development of Kodi addons";
+    license = licenses.gpl3;
```
Comment (aanderse) [inline pkgs/applications/video/kodi/addons/xbmcswift2/default.nix:22]: Sorry, I missed this first time around :disappointed: `gpl3Plus` or `gpl3Only`?

**Review by aanderse — CHANGES_REQUESTED** [review]

File: pkgs/applications/video/kodi/addons/xbmcswift2/default.nix (hunk 1 above)
Comment (dschrempf) [reply pkgs/applications/video/kodi/addons/xbmcswift2/default.nix:22]: Hm, it doesn't state anything. So I guess it is gpl3Only?

Comment (aanderse) [reply pkgs/applications/video/kodi/addons/xbmcswift2/default.nix:22]: Seems like the safe bet.

**Comment by dschrempf** [comment]
Let me know, if there is anything else I can do!

**Review by aanderse — APPROVED** [review]
Thanks!

**Comment by aanderse** [comment]
Hi @dschrempf :wave: sorry to come back to this but I was doing some `kodi` work and noticed this:

https://github.com/known-as-bmf/plugin.video.arteplussept/blob/bb3e82a7b66b4a1494e5c138a0297289d9cb1c00/addon.xml#L7

> I tested the plugins and they work on my Linux machine.

Are you sure it isn't incidental to another addon being installed that this one works? I think you need to add `dateutil` to the list of dependencies.

**Comment by dschrempf** [comment]
Hm, I tried the addon exclusively and it worked, but you are right, this dependency is listed. I added it in a separate PR but did not test the addon yet because of time constraints.

**Pull request merged**